Add twin sticks support

mistydemeo opened the pull request:
It turns out the Virtual On twin sticks are actually treated internally as a gamepad, just with an alternate control layout. Technically they could just get treated as a gamepad here, but having appropriate symbolic names to map to is a lot more convenient.

Note that this isn't the dual mission stick - that's a different device, with analogue joysticks. This is a pure digital device.

mistydemeo committed:
Add twin sticks support

It turns out the Virtual On twin sticks are actually treated internally
as a gamepad, just with an alternate control layout.
Technically they could just get treated as a gamepad here, but having
appropriate symbolic names to map to is a lot more convenient.

mistydemeo commented:
I've tested this using a branch where I've been mapping axes to buttons/d-pad directions. It's still too iffy (and the code too messy) to submit yet, but it's enough to confirm this mapping's right.

mistydemeo commented:
If you want to test this, Virtual On is the only game that uses this controller. It'll ask you what controller you're using at startup - supported in all regions.

abrasive commented:
I think I found a reasonable solution to the axis mapping issue. Using the latest master you can map analog axes to digital Controls. If you set sign=-1 you can then map the negative side of an axis to a button press, eg.

```
InputGamepad::add_axis_mapping(SDL_CONTROLLER_AXIS_LEFTX, ctl, p1->get_control("Lleft"), -1);
InputGamepad::add_axis_mapping(SDL_CONTROLLER_AXIS_LEFTX, ctl, p1->get_control("Lright"), 1);
```

The hysteresis is then adjustable per-mapping too. What do you think?

mistydemeo committed:
Add twinsticks test utility

mistydemeo commented:
That looks great! Definitely convenient, and works a treat. I do wonder whether something more readable could be passed as a `sign` value? Maybe an enum with readable symbolic names.

I've rebased on master, and added an updated test utility. This does the most natural mapping for Virtual On - one analogue stick mapped to each digital stick, left and right bumpers for the sticks' top buttons, and triggers mapped to the sticks' digital triggers.
